onnxruntime/core/session/inference_session.cc

```diff
@@ -1718,6 +1729,25 @@ common::Status InferenceSession::Run(const RunOptions& run_options, IOBinding& i
              &io_binding.GetOutputs(), &io_binding.GetOutputsDeviceInfo());
 }
 
+common::Status InferenceSession::PartialRun(const RunOptions& run_options, IOBinding& io_binding, int64_t& run_id) {
+  ORT_ENFORCE(run_options.only_execute_path_to_fetches == false);
+  ORT_ENFORCE(session_state_->GetEnableMemoryPattern() == false);
```

Comment:
>session_state_->GetEnableMemoryPattern() == false [](start = 14, length = 49)

will it impact memory footprint?

Reply:
Memory pattern generation is turned off for current OrtModule anyways, see line 385 in ortmodule.py  `session_options.enable_mem_pattern = False`

Actually, with this change we may be able to turn it on eventually. Yes, for some models that have static input shapes it can result is sub-optimal footprint.

---
In reply to: [598072745](https://github.com/microsoft/onnxruntime/pull/7077#discussion_r598072745) [](ancestors = 598072745)